lib/Conversion/TritonGPUToLLVM/Utility.cpp

```diff
@@ -231,12 +231,24 @@ emitIndicesUsingLinearLayouts(Location loc, RewriterBase &rewriter,
       withCTAOffset ? target.getClusterCTAId(rewriter, loc) : i32_val(0);
   unsigned rank = shape.size();
   SmallVector<SmallVector<Value>> ret;
+  // TODO Investigate why LLVM do no optimize one "apply" call
```

Comment:
Can we add some more comments explaining why the separation so that it gives others reading the code a better understanding? We can drop the todo here given this is a fine optimization we can do at this level--we cannot rely on llvm backend solve all the problems for us.